### ISSUE TITLE
NetworkManager isn't started automatically

### ISSUE THREAD

**Issue by ja1den** [NixOS/nixpkgs#283717]
### Describe the bug
On `nixos-unstable`, NetworkManager isn't started automatically.

### Steps To Reproduce
1. Build a system configuration with `networking.networkmanager.enable = true;`
3. Reboot.
4. `NetworkManager.service` won't be started automatically.

### Expected behavior
NetworkManager should be started automatically.

### Additional context
It seems this was introduced by 62f30634db2146c23f9aff1c9b598524652bbeb7.

If I lock `nixpkgs` at eb42a295eb59ed71727a4e144006d46628dc55b3 (the parent of 62f30634db2146c23f9aff1c9b598524652bbeb7), `NetworkManager.service` is started automatically. If I lock `nixpkgs` at 62f30634db2146c23f9aff1c9b598524652bbeb7, it isn't.

### Notify maintainers

@lf- made 62f30634db2146c23f9aff1c9b598524652bbeb7 and also #258680.
@jtojnar is the maintainer of the NetworkManager module (I think?).

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.0-zen3, NixOS, 24.05 (Uakari), 24.05.20240119.62f3063`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `""`
 - nixpkgs: `not found`
```

Note: I don't have any channels, so I ran `nix run nixpkgs#nix-info -- -m` instead.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by lf-** [comment]
Thanks for the report. We will look into this; this is not intended behaviour and suggests some service dependencies of network manager are borked and likely have been for years.

**Comment by ElvishJerricco** [comment]
Should be fixed by #283818
